plugin/advisor/mysql/advisor_naming_unique_key_convention.go

```diff
@@ -0,0 +1,184 @@
+type NamingUKConventionAdvisor struct {
+}
+
+// Check checks for index naming convention.
+func (check *NamingUKConventionAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	root, errAdvice := parseStatement(statement, ctx.Charset, ctx.Collation)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySchemaReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	format, templateList, err := api.UnmarshalNamingRulePayloadAsTemplate(ctx.Rule.Type, ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+	checker := &namingUKConventionChecker{
+		level:        level,
+		format:       format,
+		templateList: templateList,
+	}
+	for _, stmtNode := range root {
+		(stmtNode).Accept(checker)
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    common.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+	return checker.adviceList, nil
+}
+
+type namingUKConventionChecker struct {
+	adviceList   []advisor.Advice
+	level        advisor.Status
+	format       string
+	templateList []string
+}
+
+func (checker *namingUKConventionChecker) Enter(in ast.Node) (ast.Node, bool) {
+	indexDataList := checker.getMetaDataList(in)
+
+	for _, indexData := range indexDataList {
+		regex, err := getTemplateRegexp(checker.format, checker.templateList, indexData.metaData)
+		if err != nil {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    common.Internal,
+				Title:   "Internal error for unique key naming convention rule",
+				Content: fmt.Sprintf("%q meet internal error %q", in.Text(), err.Error()),
+			})
+			continue
+		}
+		if !regex.MatchString(indexData.index) {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    common.NamingUKConventionMismatch,
+				Title:   "Mismatch unique key naming convention",
+				Content: fmt.Sprintf("%q mismatches unique key naming convention, expect %q but found %q", in.Text(), checker.format, indexData.index),
+			})
+		}
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    common.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+
+	return in, false
+}
+
+func (checker *namingUKConventionChecker) Leave(in ast.Node) (ast.Node, bool) {
+	return in, true
+}
+
+// getMetaDataList returns the list of unique key with meta data.
+func (checker *namingUKConventionChecker) getMetaDataList(in ast.Node) []*indexMetaData {
+	var res []*indexMetaData
+
+	switch node := in.(type) {
+	case *ast.CreateTableStmt:
+		for _, constraint := range node.Constraints {
+			switch constraint.Tp {
+			case ast.ConstraintUniq, ast.ConstraintUniqKey, ast.ConstraintUniqIndex:
+				var columnList []string
```

Comment:
ditto
use `if` instead of `switch case`

Reply:
We have three case here, so I'd prefer using `case x1, x2, x3` instead of `if tp == x1 || tp == x2 || tp== x3`